osu.Game/Graphics/UserInterface/Bar.cs

```diff
@@ -0,0 +1,134 @@
+﻿// Copyright (c) 2007-2017 ppy Pty Ltd <contact@ppy.sh>.
+// Licensed under the MIT Licence - https://raw.githubusercontent.com/ppy/osu/master/LICENCE
+
+using OpenTK;
+using OpenTK.Graphics;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Sprites;
+using System;
+
+namespace osu.Game.Graphics.UserInterface
+{
+    public class Bar : Container, IHasAccentColour
+    {
+        private Box background;
+        private readonly Box bar;
+
+        private const int resize_duration = 250;
+
+        private const EasingTypes easing = EasingTypes.InOutCubic;
+
+        private float length;
+        public float Length
+        {
+            get
+            {
+                return length;
+            }
+            set
+            {
+                length = MathHelper.Clamp(value, 0, 1);
+                updateBarLength();
+            }
+        }
+
+        public Color4 BackgroundColour
+        {
+            get
+            {
+                return background?.Colour ?? default(Color4);
+            }
+            set
+            {
+                if (background == null)
+                    Add(background = new Box

```

Comment:
Does the framework render something with a transparent color? IsPresent for example only checks Alpha